[2024-06-18T07:07:46.028+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-18T07:07:46.130+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: send_message_and_calculate_ABC.send_message_and_calculate_ABC scheduled__2024-06-17T07:00:00+00:00 [queued]>
[2024-06-18T07:07:46.161+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: send_message_and_calculate_ABC.send_message_and_calculate_ABC scheduled__2024-06-17T07:00:00+00:00 [queued]>
[2024-06-18T07:07:46.161+0000] {taskinstance.py:2303} INFO - Starting attempt 4 of 6
[2024-06-18T07:07:46.233+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): send_message_and_calculate_ABC> on 2024-06-17 07:00:00+00:00
[2024-06-18T07:07:46.261+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'send_message_and_calculate_ABC', 'send_message_and_calculate_ABC', 'scheduled__2024-06-17T07:00:00+00:00', '--job-id', '1613', '--raw', '--subdir', 'DAGS_FOLDER/send_message_and_calculate_ABC.py', '--cfg-path', '/tmp/tmprk34wz9b']
[2024-06-18T07:07:46.265+0000] {standard_task_runner.py:91} INFO - Job 1613: Subtask send_message_and_calculate_ABC
[2024-06-18T07:07:46.266+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=102710) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-18T07:07:46.268+0000] {standard_task_runner.py:63} INFO - Started process 102712 to run task
[2024-06-18T07:07:46.591+0000] {task_command.py:426} INFO - Running <TaskInstance: send_message_and_calculate_ABC.send_message_and_calculate_ABC scheduled__2024-06-17T07:00:00+00:00 [running]> on host d40bdb4a74ef
[2024-06-18T07:07:46.948+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='send_message_and_calculate_ABC' AIRFLOW_CTX_TASK_ID='send_message_and_calculate_ABC' AIRFLOW_CTX_EXECUTION_DATE='2024-06-17T07:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-06-17T07:00:00+00:00'
[2024-06-18T07:07:46.949+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-18T07:07:51.877+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-18T07:07:51.879+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/send_message_and_calculate_ABC.py", line 639, in start_dag
    main()
  File "/opt/airflow/dags/send_message_and_calculate_ABC.py", line 629, in main
    categories = calculate_ABC(vendor_codes_by_period, vendorCode_dict, categories_by_period)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/send_message_and_calculate_ABC.py", line 531, in calculate_ABC
    ABC_14(vendor_codes_by_period[1]),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/send_message_and_calculate_ABC.py", line 445, in ABC_14
    elif turnover[ven] <= CONFIG_ABC['EBITDA']["ABC_TURNOVER_14"]:
                          ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
KeyError: 'ABC_TURNOVER_14'
[2024-06-18T07:07:52.057+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=send_message_and_calculate_ABC, task_id=send_message_and_calculate_ABC, execution_date=20240617T070000, start_date=20240618T070746, end_date=20240618T070752
[2024-06-18T07:07:52.232+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 1613 for task send_message_and_calculate_ABC ('ABC_TURNOVER_14'; 102712)
[2024-06-18T07:07:52.289+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-18T07:07:52.408+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-18T07:07:52.418+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
